Parse active connection information

chenj47 opened the pull request:
Several key features involved in this pull request:
1. Create the proto buff file that defines the ActiveConn (Active Connection) Information.
2. Update the script to generate the pb.go files.
3. Implement the basic function to parse active connection information and store the results using proto buff. Also, add several test cases for the active connection part.

Note that the usage of proto buff is just a tryout in this PR since I want to see if proto buff can be easily used to send data to the front end. If it does not work well, the information will be stored in CSV then.

chenj47 committed:
Fix a bug in old Battery Historian repo + Rename the battery-historian file to sensor-historian + Get full information for each sensor

chenj47 committed:
Fix the comments in the presenter.go file + Use one expression to match the DUMP OF SERVICE line and check for substring "sensorservice"

chenj47 committed:
Stick with the old SensorInfo struct in the map instead of the pointer

chenj47 committed:
Added Comments + Accomodate the sensor list format for Android version from MNC or before

chenj47 committed:
Add the testcase to accommodate sensor Number being 00000000

chenj47 committed:
Parse the minRate/maxRate or maxDelay/minDelay values and store the delay in us rather than rate in Hz. Also accommodate this feature to older Android Version

chenj47 committed:
Fix spacing

chenj47 committed:
Add comments and fix spacing in the bugreportutils/ and historianutils/ folders
Add more comments to explain the regular expressions used.
Fix the spacing in some of the functions so that each chunk of code is doing one thing.
Reformat some comments so that most of the lines are under 100 characters. Note that the traditional 80 characters limit is not adopted since many lines of code are calling functions from other file so the long package name is unevitable. Some lines that involve if statements are not strictly under 100 characters because of Golang convention.

chenj47 committed:
Parse active connection information

chenj47 committed:
Parse active connection information

chenj47 committed:
update the vairable names for some regular expressions + update the type defined for the activeSensors map to avoid using uid

chenj47 committed:
update the vairable names for some regular expressions + update the type defined for the activeSensors map to avoid using uid

agarwal-v commented on regen_proto.sh:
```diff
@@ -19,8 +19,9 @@
-protoc --go_out=. github.com/googleinterns/sensor-historian/pb/usagestats_proto/*.proto
-protoc --go_out=. github.com/googleinterns/sensor-historian/pb/session_proto/*.proto
\ No newline at end of file
+protoc -I=github.com/googleinterns/sensor-historian/pb/activeconnection_proto/ --go_out=. github.com/googleinterns/sensor-historian/pb/activeconnection_proto/*.proto
```
Is -I flag needed?

agarwal-v commented on sensorserviceutils/sensorserviceutils.go:
```diff
@@ -0,0 +1,376 @@
+)
+
+var (
+	// Each of the connectionLineXRE is a regular expression to match the X line of
```
connLineXRE

agarwal-v commented on sensorserviceutils/sensorserviceutils.go:
```diff
@@ -0,0 +1,376 @@
+	sensors map[int32]bugreportutils.SensorInfo
+	// activeConns is a map from the active connection number to the corresponding connection info.
+	activeConns map[int32]*acpb.ActiveConn
+	// activeSensors is a map from sensor number to
```
Reword comment

agarwal-v commented on sensorserviceutils/sensorserviceutils.go:
```diff
@@ -0,0 +1,376 @@
+	// activeSensors is a map from sensor number to
+	// a map from packageName to the active connection number.
+	activeSensors map[int32]map[string]int32
+	// historyByUID is a map from a key to SubscriptionInfo.
```
history or historybyUID?

agarwal-v commented on sensorserviceutils/sensorserviceutils.go:
```diff
@@ -0,0 +1,376 @@
+			}
+			continue
+		}
+		// Previous registration parising.
```
Remove this code as it is not being usedright now.

agarwal-v reviewed (commented):
Need to rebase on previous pull request to be able to focus on changes in this request.

chenj47 committed:
Merge pull request #4 from googleinterns/parse-sensorservice

Parse full information for each sensor

chenj47 committed:
Update the script to generate proto buff related file: the I flag is not needed

chenj47 committed:
Redefine the maps used to store information and fix spacing and comments
The new maps all use the identifier for information look-up. Using the same identifier format ensures code clarity.
With the new maps define, rewrite the comments for these variables with clearer explanation.
Also rebase the code on the most up-to-date master branch

chenj47 committed:
Resolve merge conflicts

chenj47 commented:
Closing this PR to create a new one that better focus on the change for parse-activeConn branch.
